rre-dataset-generator/src/search_engine/data_store.py

```
@@ -0,0 +1,115 @@
+from __future__ import annotations
+
+import logging
+import uuid
+from typing import Dict, Tuple, List
+
+from src.logger import configure_logging
+
+configure_logging(level=logging.INFO)
+log = logging.getLogger(__name__)
+
+
+class DataStore:
+    """
+    Public API for storing, managing documents, queries, and relevance scores.
+    """
+
+    def __init__(self):
+        self.documents: Dict[str, str] = {}  # doc_id → document text
+        self.queries: Dict[str, QueryObject] = {}  # query_id → QueryObject
+
+    def _get_query_object(self, query_id: str) -> QueryObject:
+        if query_id not in self.queries:
+            log.error("Query id %s not found in DataStore", query_id)
+            raise KeyError(f"Query id '{query_id}' not found in DataStore")
+        return self.queries[query_id]
+
+    def _get_document(self, doc_id: str) -> str:
+        if doc_id not in self.documents:
+            log.error("Document id %s not found in DataStore", doc_id)
+            raise KeyError(f"Document id '{doc_id}' not found in DataStore")
+        return self.documents[doc_id]
+
+    def add_document(self, doc_id: str, document: str) -> None:
+        self.documents[doc_id] = document
```

It doesn't check if the document ID already exists.
I would raise an exception if the document already exists. If we need to update a document in the future, we can create another method to update it explicitly

---

updated